dapr/clients/grpc/client.py

```diff
@@ -451,6 +455,56 @@ def save_state(
         return DaprResponse(
             headers=call.initial_metadata())
 
+    def execute_transaction(
+            self,
+            store_name: str,
+            operations: Sequence[TransactionalStateOperation],
+            metadata: Optional[MetadataTuple] = ()) -> DaprResponse:
+        """Saves or deletes key-value pairs to a statestore as a transaction
+
+        This saves or deletes key-values to the statestore as part of a single transaction,
+        metadata can be passed with metadata field.
+
+        The example saves states to a statestore:
+            from dapr import DaprClient
+            with DaprClient() as d:
+                resp = d.execute_transaction(
+                    store_name='state_store',
+                    operations=[
+                        TransactionalStateOperation(key=key, data=value),
+                        TransactionalStateOperation(key=another_key, data=another_value),
+                    ],
+                    metadata=(
+                        ('header1', 'value1')
+                    ),
+                )
+
+        Args:
+            store_name (str): the state store name to save to.
+            operations (Sequence[TransactionalStateOperation]): the transaction operations.
+            metadata (tuple, optional): custom metadata.
+
+        Returns:
+            :class:`DaprResponse` gRPC metadata returned from callee
+        """
+        if not store_name or len(store_name) == 0 or len(store_name.strip()) == 0:
+            raise ValueError("State store name cannot be empty")
+        req_ops = [api_v1.TransactionalStateOperation(
+            operationType=o.operation_type.value,
+            request=common_v1.StateItem(
+                key=o.key,
+                value=to_bytes(o.data),
+                etag=o.etag,
+                metadata=o.metadata)) for o in operations]
```

Comment:
we do not support per operation metadata for transactional request.

Reply:
We do, it is in the proto.